Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterAppState.kt

```diff
@@ -58,6 +59,25 @@ class JetcasterAppState(
         isOnline = checkIfOnline()
     }
 
+    fun navigateToPlayer(episodeUri: String) {
+        val newUri = episodeUri.replace("/", "|")
+        navController.navigate(Screen.Player.createRoute(newUri))
+    }
+
+    fun navigateBack() {
+        navController.popBackStack()
+    }
+
+    /**
+     * Revert back the characters that [navigateToPlayer] changed to make the argument compatible
+     * with Navigation.
+     */
+    fun decryptPlayerArgumentUri(bundle: Bundle?) {
```

Comment:
Can we use the standard `Uri.encode`/`Uri.decode` pair of functions here? I'd also prefer if we just did that in `HomeViewModel` or passed in the argument directly, versus rewriting a `Bundle`, which seems like it could be problematic.

Reply:
Good point! Done, thanks!